statefun-flink/statefun-flink-core/src/main/java/org/apache/flink/statefun/flink/core/httpfn/HttpFunctionSpec.java

```diff
@@ -111,14 +138,36 @@ public Builder withMaxRequestDuration(Duration duration) {
       return this;
     }
 
+    public Builder withConnectTimeoutDuration(Duration duration) {
+      this.connectTimeout = Objects.requireNonNull(duration);
+      return this;
+    }
+
+    public Builder withReadTimeoutDuration(Duration duration) {
+      this.readTimeout = Objects.requireNonNull(duration);
+      return this;
+    }
+
+    public Builder withWriteTimeoutDuration(Duration duration) {
+      this.writeTimeout = Objects.requireNonNull(duration);
+      return this;
+    }
+
     public Builder withMaxNumBatchRequests(int maxNumBatchRequests) {
       this.maxNumBatchRequests = maxNumBatchRequests;
       return this;
     }
 
     public HttpFunctionSpec build() {
       return new HttpFunctionSpec(
-          functionType, endpoint, states, maxRequestDuration, maxNumBatchRequests);
+          functionType,
```

Comment:
I think that somewhere we should validate the timeouts that they make sense.
For example: all the durations are non zero, and the sum of read write and connect is <= call timeout.

Reply:
0 duration would mean that no timeout is imposed. Should that really be an invalid configuration?